components/eamxx/src/share/field/field_manager.cpp

```diff
@@ -117,109 +129,146 @@ add_to_group (const std::string& field_name, const std::string& group_name)
   return *ptr;
 }
 
-Field& FieldManager::get_field (const std::string& name) {
+Field& FieldManager::get_field (const std::string& name, const std::string& grid_name) {
 
   EKAT_REQUIRE_MSG(m_repo_state==RepoState::Closed,
       "Error! Cannot get fields from the repo while registration has not yet completed.\n");
-  auto ptr = get_field_ptr(name);
-  EKAT_REQUIRE_MSG(ptr!=nullptr, "Error! Field " + name + " not found.\n");
+  auto ptr = get_field_ptr(name, grid_name);
+  EKAT_REQUIRE_MSG(ptr!=nullptr, "Error! Field " + name + " on grid " + grid_name + " not found.\n");
   return *ptr;
 }
 
 FieldGroup FieldManager::
-get_field_group (const std::string& group_name) const
+get_field_group (const std::string& group_name, const std::string& grid_name) const
 {
   // Sanity checks
   EKAT_REQUIRE_MSG(m_repo_state==RepoState::Closed,
       "Error! Cannot get field groups from the repo while registration has not yet completed.\n");
-  EKAT_REQUIRE_MSG (has_group(group_name),
-      "Error! Field group '" + group_name + "' not found.\n");
+  EKAT_REQUIRE_MSG (has_group(group_name, grid_name),
+      "Error! Field group '" + group_name + "' on grid '" + grid_name + "' not found.\n");
 
   // Create an empty group
   FieldGroup group(group_name);
 
   // Set the info in the group
   group.m_info = m_field_groups.at(group_name);
 
-  // IMPORTANT: this group might have been created as a "Copy" (see field_request.hpp)
-  //            of another group. In this case, we only allocate the bundled group,
-  //            and we don't create individual subfields. Users can still extract
-  //            subfields on the fly, if need be, by using the names/idx data in m_info.
-  auto it = m_group_requests.find(group_name);
-  bool allocate_subfields = true;
-  if (it!=m_group_requests.end()) {
-    for (const auto& req : it->second) {
-      if (req.derived_type==DerivationType::Copy) {
-        allocate_subfields = false;
-        break;
-      }
-    }
-  }
-
-  if (allocate_subfields) {
-    // Find all the fields and set them in the group
-    for (const auto& fname : group.m_info->m_fields_names) {
-      auto f = get_field_ptr(fname);
-      group.m_fields[fname] = f;
-    }
+  // Find all the fields and set them in the group
+  for (const auto& fname : group.m_info->m_fields_names) {
+    auto f = get_field_ptr(fname, grid_name);
+    group.m_fields[fname] = f;
   }
 
   // Fetch the bundle field (if bundled)
   if (group.m_info->m_bundled) {
-    if (allocate_subfields) {
-      // We can get the parent from any of the fields in the group.
-      auto p = group.m_fields.begin()->second->get_header().get_parent().lock();
-      EKAT_REQUIRE_MSG(p!=nullptr,
-          "Error! A field belonging to a bundled field group is missing its 'parent'.\n");
-
-      const auto& p_id = p->get_identifier();
-      group.m_bundle = get_field_ptr(p_id);
+    // All fields in a group have the same parent, get the parent from the 1st one
```

Comment:
The following is the logic that allows us to detect if this is the parent group or a sub group, and getting the correct slice indices.